javatests/arcs/core/allocator/AllocatorTestBase.kt

```diff
@@ -96,31 +87,15 @@ open class AllocatorTestBase {
         hostRegistry = hostRegistry()
         allocator = Allocator.create(hostRegistry, TimeImpl())
 
-        recipePersonStorageKey = CreateableStorageKey(
-            "recipePerson", storageCapability
-        )
-        writePersonHandleConnection =
-            Plan.HandleConnection(recipePersonStorageKey, HandleMode.Write, personEntityType)
-
-        writePersonParticle = Plan.Particle(
-            "WritePerson", WritePerson::class.java.getCanonicalName()!!,
-            mapOf("person" to writePersonHandleConnection)
-        )
-
-        readPersonHandleConnection = Plan.HandleConnection(
-            recipePersonStorageKey,
-            HandleMode.Read,
-            personEntityType
-        )
-
-        readPersonParticle = Plan.Particle(
-            "ReadPerson", ReadPerson::class.java.getCanonicalName()!!,
-            mapOf("person" to readPersonHandleConnection)
-        )
+        readPersonParticle =
+            requireNotNull(PersonRecipePlan.particles.find { it.particleName == "ReadPerson" }) {
```

Comment:
Is this a common type of query? Or just something needed for testing?

Reply:
Given that the order of r2p is unpredictable, this is how I pull out the Plan.Particle representing the ReadPerson part.  This is just needed for testing.